presto-plugin-toolkit/src/test/java/io/prestosql/plugin/base/util/TestJsonUtils.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.base.util;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+
+import static io.prestosql.plugin.base.util.JsonUtils.parseJson;
+import static io.prestosql.plugin.base.util.TestJsonUtils.TestEnum.OPTION_A;
+import static java.nio.charset.StandardCharsets.US_ASCII;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestJsonUtils
+{
+    enum TestEnum
+    {
+        OPTION_A,
+        OPTION_B
+    }
+
+    public static class TestObject
+    {
+        @JsonProperty
+        private TestEnum testEnum;
+    }
+
+    @Test
+    public void testLowercaseEnum()
+            throws IOException
+    {
+        TestObject parsed = parseJson("{\"testEnum\": \"option_a\"}".getBytes(US_ASCII), TestObject.class);
```

Comment:
please also test mixed and upper case.

Reply:
you told that we should support only lower case

uppercase/mixed case would also work (there's no switch in Jackson to enable only lowercase, there's only `ACCEPT_CASE_INSENSITIVE_ENUMS`), but I would say it should stay undocumented, untested and unsupported, if somebody uses it and some day it stops working, it's their fault